src/providers/DevicesProvider/AudioInputProvider.tsx

```diff
@@ -104,11 +104,20 @@ const AudioInputProvider: React.FC = ({ children }) => {
       }
     }
 
+    const callback = (deviceLabels: DeviceLabels): void => {
+      if (deviceLabels === DeviceLabels.Audio || deviceLabels === DeviceLabels.AudioAndVideo) {
+        initAudioInput();
+      }
+    };
+
+    meetingManager.subscribeToDeviceLabelsChange(callback);
+
     initAudioInput();
```

Comment:
Isn't this incorrect? We always call initAudioInput() same as before?

Reply:
We control the device permission request by `deviceLabelTrigger`. We setup the `deviceLabelTrigger` before the first time `initAUdioInput()` is called.